tests/plugins/test_tag_by_labels.py

```diff
@@ -68,8 +68,14 @@ def test_tag_by_labels_plugin(tmpdir):
     assert TagByLabelsPlugin.key in output.keys()
     assert len(workflow.tag_conf.images) == 4
     images = [i.to_str() for i in workflow.tag_conf.images]
+    primary_images = [i.to_str() for i in workflow.tag_conf._primary_images]
```

Comment:
So, really this was found by coveralls, but it took quite a bit of digging. Although it tells you the relative increase/decrease for coverage, it won't tell you which lines cause the change! You have to compare with the base version side-by-side.

That gets you as far as finding out which bit isn't tested, so then it's a case of reading through the code to find out why it isn't, which lead me here.